### PR TITLE
Add basic styling to user profile page

### PR DESCRIPTION
**Pull request by thestrabusiness**
This commit adds some nicer styling to the user profile page, including
adding the user's avatar to the page.

We also refactored the `UserAvatar` class to accept a size argument so a
larger-size image can be used where desired. These changes include some
clean up of the tests for the `UserAvatar` class.

Screen Shot 2020-12-08 at 11 04 57 AM

Co-authored-by: Kurt G. <soad1789@gmail.com>

### CODE REVIEW

File: app/renderers/user_avatar.rb [hunk 1]
```diff
@@ -4,32 +4,52 @@ class UserAvatar
   include ActionView::Helpers::AssetTagHelper
   include ActionView::Helpers::TagHelper
 
-  def initialize(user)
+  class SizeNotDefined < StandardError; end
+
+  AVATAR_SIZES = [:small, :large].freeze
+  LARGE_SIZE = 250
```
Comment (americanhanko) [inline app/renderers/user_avatar.rb:10]: The name is a bit ambiguous. Maybe we could rename this to be `AVATAR_LARGE_SIZE_PIXELS`?

File: app/views/users/show.html.erb [hunk 2]
```diff
@@ -1,16 +1,19 @@
...
-      <%= link_to "Edit Profile", edit_user_path %>
-    </div>
+
+    <%= UserAvatar.render(current_user, size: :large) %>
+
+    <div class="subtitle">Email: <%= current_user.email %></div>
+    <% if current_user.receive_reminder_email %>
+      <div class="subtitle">You are subscribed to reminder emails</div>
+    <% else %>
+      <div class="subtitle">You are not subscribed to reminder emails</div>
```
Comment (americanhanko) [inline app/views/users/show.html.erb:16]: Is the user able to easily opt-in or opt-out?

File: spec/renderers/user_avatar_spec.rb [hunk 3]
```diff
@@ -3,27 +3,95 @@
...
+        user = build(:user)
+        image_url = build_gravatar_url(user)
+        gravatar_size = UserAvatar::LARGE_SIZE
+
+        result = UserAvatar.render(user, size: :large)
+
+        expect(result).to include "src=\"#{image_url}?s=#{gravatar_size}\""
+      end
+    end
+  end
```
Comment (americanhanko) [inline spec/renderers/user_avatar_spec.rb:79]: Could we add a test to ensure we raise `SizeNotDefined` if the user's gravatar doesn't meet our requirements?

**Review by americanhanko — DISMISSED** [review]
Looks good 👍  Approved with a few suggestions.

File: app/views/users/show.html.erb (hunk 2 above)
Comment (thestrabusiness) [reply app/views/users/show.html.erb:16]: Yep, This was a feature recently added in

File: spec/renderers/user_avatar_spec.rb (hunk 3 above)
Comment (thestrabusiness) [reply spec/renderers/user_avatar_spec.rb:79]: Yup! There's already such a test at the top of the file.

File: app/renderers/user_avatar.rb (hunk 1 above)
Comment (thestrabusiness) [reply app/renderers/user_avatar.rb:10]: Hmm yeah. Maybe `LARGE_PIXEL_SIZE`? Now that I'm looking at this, I wonder if the `AVATAR_` prefix is redundant, especially from the POV of this namespace. I wonder if these work better like this:
```ruby
UserAvatar::SIZES
UserAvatar::LARGE_PIXEL_SIZE
```

**Pull request merged**